pkg/promtail/target.go

```diff
@@ -55,14 +55,32 @@ type Target struct {
 
 // NewTarget create a new Target.
 func NewTarget(logger log.Logger, handler EntryHandler, positions *Positions, path string, labels model.LabelSet) (*Target, error) {
+	var err error
+	path, err = filepath.Abs(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "filepath.Abs")
+	}
+	matches, err := filepath.Glob(path)
```

Comment:
the `nil` wont cause any problems as all we do with `matches` is loop through the values, which is safe to do.
But i think we do need to handle this situation better.  Right now, if nothing matches we will launch the tailer goroutine via `t,run()` but it wont ever do anything as there are no files being tailed and no directories being watched for new files that might be created.

I dont think we should return an error here.  I think the correct thing to do is periodically call `filepath.Glob(path)` so that if any new files/directories do get created we start tailing them.

Reply:
Sounds good to me, part of what @tomwilkie and I talked about was adding a process to periodically check for new files/changes in case we miss a fsnotify event.  I think this can be addressed by that work